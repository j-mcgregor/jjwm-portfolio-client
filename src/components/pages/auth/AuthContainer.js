/* eslint-disable operator-linebreak */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import LoginSection from './sections/Login';
import RegisterSection from './sections/Register';
import S from '../../styledComponents';

const AuthContainer = ({ loginUser, registerUser, auth }) => {
  const [isLogin, setLogin] = useState(true);

  const { errors } = auth;
  const networkErr =
    errors &&
    (errors.login === 'Network Error' || errors.register === 'Network Error');

  return (
    <S.AuthContainer>
      <S.AuthHeader isLogin={isLogin}>
        <ul>
          <li className="login" onClick={() => setLogin(true)}>
            {' '}
            <span>Login</span>
          </li>
          <li className="register" onClick={() => setLogin(false)}>
            {' '}
            <span>Register</span>
          </li>
        </ul>
      </S.AuthHeader>
      {isLogin ? (
        <LoginSection
          onSubmit={loginUser}
          networkErr={networkErr}
          errors={errors && errors.login}
        />
      ) : (
        <RegisterSection
          onSubmit={registerUser}
          networkErr={networkErr}
          errors={errors && errors.registration}
        />
      )}
      {networkErr && <S.AuthFooter>Network Error</S.AuthFooter>}
    </S.AuthContainer>
  );
};

AuthContainer.propTypes = {
  loginUser: PropTypes.func,
  registerUser: PropTypes.func,
  auth: PropTypes.shape({
    errors: PropTypes.shape({})
  }).isRequired
};

AuthContainer.defaultProps = {
  loginUser: () => {},
  registerUser: () => {}
};

export default AuthContainer;
