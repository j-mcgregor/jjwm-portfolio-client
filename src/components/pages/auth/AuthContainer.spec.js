import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AuthContainer from './AuthContainer';

Enzyme.configure({ adapter: new Adapter() });

describe('AuthContainer', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      loginUser: jest.fn(),
      registerUser: jest.fn(),
      auth: {
        errors: {}
      }
    };
    wrapper = shallow(<AuthContainer {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
