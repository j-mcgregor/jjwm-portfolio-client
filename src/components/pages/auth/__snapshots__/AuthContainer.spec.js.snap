// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthContainer should render correctly 1`] = `ShallowWrapper {}`;
